src/Features/CSharp/Portable/CodeRefactorings/NamedArguments/CSharpAddNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,188 @@
+                    return false;
+            }
+        }
+
+        protected override bool TryGetArguments(SyntaxNode node, SemanticModel semanticModel, out IList<ArgumentSyntax> arguments)
+        {
+
+            if (node == null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            if (semanticModel == null)
+            {
+                throw new ArgumentNullException(nameof(semanticModel));
+            }
+
+            switch (node.Kind())
+            {
+                case SyntaxKind.InvocationExpression:
+                    {
+                        var syntaxNode = (InvocationExpressionSyntax)node;
+
+                        // Skip nameof
+                        var symbol = semanticModel.GetSymbolInfo(syntaxNode).Symbol;
+                        if (symbol == null)
+                        {
+                            goto default;
+                        }
+
+                        arguments = syntaxNode.ArgumentList.Arguments.ToList();
+                        return true;
+                    }
+
+                case SyntaxKind.ElementAccessExpression:
+                    {
+                        var syntaxNode = (ElementAccessExpressionSyntax)node;
+
+                        // Array access is not allowed to contain named arguments
+                        var type = semanticModel.GetTypeInfo(syntaxNode.Expression).Type;
+                        if (type.TypeKind == TypeKind.Array)
+                        {
+                            goto default;
+                        }
+
+                        arguments = syntaxNode.ArgumentList.Arguments.ToList();
+                        return true;
+                    }
+
+                case SyntaxKind.ObjectCreationExpression:
+                    {
+                        var syntaxNode = (ObjectCreationExpressionSyntax)node;
+                        arguments = syntaxNode.ArgumentList.Arguments.ToList();
+                        return true;
+                    }
+
+                case SyntaxKind.BaseConstructorInitializer:
+                case SyntaxKind.ThisConstructorInitializer:
+                    {
+                        var syntaxNode = (ConstructorInitializerSyntax)node;
+                        arguments = syntaxNode.ArgumentList.Arguments.ToList();
+                        return true;
+                    }
+
+                default:
+                    arguments = null;
+                    return false;
+            }
+        }
+
+        protected override SyntaxNode WithArguments(SyntaxNode node, IEnumerable<ArgumentSyntax> arguments)
+        {
+
+            if (node == null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            if (arguments == null)
+            {
+                throw new ArgumentNullException(nameof(arguments));
+            }
+
+            var argumentList = SyntaxFactory.SeparatedList(arguments);
```

Comment:
This will lose all the trivia of hte prevous separated list.  We should really be just taking the previous argument list node and replacing the individual subnodes in it for maximum compat with the original user code.